json_serializable/test/integration/json_keys_example.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'json_keys_example.g.dart';
+
+@JsonSerializable(createJsonKeys: true, fieldRename: FieldRename.kebab)
+class Model {
+  Model({
+    required this.firstName,
+    required this.lastName,
+    this.ignoredName,
+  });
+
+  factory Model.fromJson(Map<String, Object?> json) => _$ModelFromJson(json);
+
+  final String firstName;
+
+  @JsonKey(name: 'LAST_NAME')
+  final String lastName;
+
+  @JsonKey(includeFromJson: false, includeToJson: false)
+  final String? ignoredName;
+
+  String get fullName => '$firstName $lastName';
+
+  Map<String, Object?> toJson() => _$ModelToJson(this);
+
+  static const jsonKeys = _$ModelJsonKeys();
+}
+
+@JsonSerializable(
+  createJsonKeys: true,
+  fieldRename: FieldRename.kebab,
+  createFactory: false,
+)
+class _PrivateModel {
+  _PrivateModel(this.fullName);
+
+  final String fullName;
+
+  Map<String, Object?> toJson() => _$PrivateModelToJson(this);
+
+  // ignore: unused_field
+  static const jsonKeys = _$PrivateModelJsonKeys();
```

Comment:
you could add a test for this! Would be a great way to get rid of the lint!

Reply:
You're right, I added some integration tests